airbyte-ci/connectors/pipelines/tests/test_helpers/test_run_steps.py

```diff
@@ -0,0 +1,238 @@
+import anyio
+import pytest
+import time
+from pipelines.models.contexts import PipelineContext
+from pipelines.models.steps import Step, StepResult, StepStatus
+from pipelines.helpers.run_steps import RunStepOptions, StepToRun, run_steps
+
+test_context = PipelineContext(
+    pipeline_name="test",
+    is_local=True,
+    git_branch="test",
+    git_revision="test"
+)
+
+class TestStep(Step):
+    title = "Test Step"
+    async def _run(self, result_status = StepStatus.SUCCESS) -> StepResult:
+        return StepResult(self, result_status)
+
+@pytest.mark.anyio
+@pytest.mark.parametrize("steps, expected_results, options", [
+    (
+        [
+            [StepToRun(id="step1", step=TestStep(test_context))],
+            [StepToRun(id="step2", step=TestStep(test_context))],
+            [StepToRun(id="step3", step=TestStep(test_context))],
+            [StepToRun(id="step4", step=TestStep(test_context))],
+        ],
+        {
+            "step1": StepStatus.SUCCESS,
+            "step2": StepStatus.SUCCESS,
+            "step3": StepStatus.SUCCESS,
+            "step4": StepStatus.SUCCESS
+        },
+        RunStepOptions(fail_fast=True)
```

Comment:
```suggestion
       id="Something meaningful describing the test case"
        [
            [StepToRun(id="step1", step=TestStep(test_context))],
            [StepToRun(id="step2", step=TestStep(test_context))],
            [StepToRun(id="step3", step=TestStep(test_context))],
            [StepToRun(id="step4", step=TestStep(test_context))],
        ],
        {
            "step1": StepStatus.SUCCESS,
            "step2": StepStatus.SUCCESS,
            "step3": StepStatus.SUCCESS,
            "step4": StepStatus.SUCCESS
        },
        RunStepOptions(fail_fast=True)
```
Would be helpful to grasp what's tested or not

Reply:
Good call!